### ISSUE TITLE
Intermitten ERROR:  While executing gem ... (NoMethodError)     undefined method `name' for nil:NilClass

### ISSUE THREAD

**Issue by penchef** [maxim-lobanov/setup-cocoapods#20]
Hey @maxim-lobanov ! Thanks for your action, real nice!

We've seen some intermittent failures, which I can't figure out why they happen. Do you have any clue what's going on there?

Intermittent Failure:
```log
Run maxim-lobanov/setup-cocoapods@v1
Podfile points to the Cocoapods 1.10.2
/usr/local/lib/ruby/gems/2.7.0/bin/pod --version
1.11.2
/usr/local/opt/ruby@2.7/bin/gem uninstall cocoapods --all --executables
/usr/local/opt/ruby@2.7/bin/gem install cocoapods -v 1.10.2 --no-document
Removing pod
Removing sandbox-pod
Successfully uninstalled cocoapods-1.11.2
ERROR:  While executing gem ... (NoMethodError)
    undefined method `name' for nil:NilClass
Error: The process '/usr/local/opt/ruby@2.7/bin/gem' failed with exit code 1
```

This is our yml:
 
```yml
name: CD
 
jobs:
  build_release_ios:
    name: Release
    runs-on: macos-10.15
       steps:
....
      - name: Setup Cocoapods
        uses: maxim-lobanov/setup-cocoapods@v1
        with:
          podfile-path: ios/Podfile.lock

```

Podfile.lock: `COCOAPODS: 1.10.2`

A parallel job (executing the same code, just for another whitelabel flavor) is sucessfull
```log
Run maxim-lobanov/setup-cocoapods@v1
Podfile points to the Cocoapods 1.10.2
/usr/local/lib/ruby/gems/2.7.0/bin/pod --version
1.11.2
/usr/local/opt/ruby@2.7/bin/gem uninstall cocoapods --all --executables
/usr/local/opt/ruby@2.7/bin/gem install cocoapods -v 1.10.2 --no-document
Removing pod
Removing sandbox-pod
Successfully uninstalled cocoapods-1.11.2
Successfully installed ruby-macho-1.4.0
Successfully installed molinillo-0.6.6
Successfully installed thread_safe-0.3.6
Successfully installed tzinfo-1.2.9
Successfully installed activesupport-5.2.6
Successfully installed cocoapods-core-1.10.2
Successfully installed cocoapods-1.10.2
7 gems installed
Cocoapods 1.10.2 has been installed successfully
```

**Comment by dmeyer-pfg** [comment]
I am seeing this issue intermittently as well. The issue appears to be due a race condition where the async command to remove the existing version of cocoapods has not completed prior to attempting to install the version specified by the user. For reference, [here](https://github.com/maxim-lobanov/setup-cocoapods/blob/master/src/installer.ts#L15-L19) is the code I am referring to. I believe this change should be made:


```javascript
- console.log('et');
+ console.log('g');

**Comment by dmeyer-pfg** [comment]
I am seeing this issue intermittently as well. The issue appears to be due a race condition where the async command to remove the existing version of cocoapods has not completed prior to attempting to install the version specified by the user. For reference, [here](https://github.com/maxim-lobanov/setup-cocoapods/blob/master/src/installer.ts#L15-L19) is the code I am referring to. This is the change that should be made. I'll open a PR.


```diff
+ await exec.exec("gem", ["uninstall", "cocoapods", "--all", "--executables", "--ignore-dependencies"]);
- exec.exec("gem", ["uninstall", "cocoapods", "--all", "--executables", "--ignore-dependencies"]);
```

**Comment by xxfast** [comment]
Running into this more frequently [here](https://github.com/xxfast/NYTimes-KMP/actions/runs/5183819176/jobs/9342129694)

**Comment by maxim-lobanov** [comment]
Thank you for contribution and fixing this issue!
I have released `v1.4.0` version with fix and updated `v1` tag
